source/common/http/conn_manager_impl.cc

```diff
@@ -814,7 +816,9 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
     // HTTP/1.1.
     //
     // The protocol may have shifted in the HTTP/1.0 case so reset it.
-    stream_info_.protocol(protocol);
+    if (Http::Utility::getProtocol(stream_info_.protocols()) != protocol) {
+      stream_info_.addProtocol(Http::Utility::getProtocolString(protocol));
```

Comment:
This is going to change the behavior, right? Meaning we will now have multiple HTTP protocol strings?

Reply:
It is always appended and `getProtocol` looks from the back, so there is no change in the behavior. But we will have multiple strings in the protocol list as decisions are made about the protocol.